android/src/main/java/com/bitmovin/player/reactnative/RNBasePlayerView.kt

```diff
@@ -0,0 +1,241 @@
+package com.bitmovin.player.reactnative
+
+import android.annotation.SuppressLint
+import android.app.PictureInPictureParams
+import android.content.pm.PackageManager
+import android.content.res.Configuration
+import android.graphics.Rect
+import android.os.Build
+import android.util.Rational
+import android.view.View
+import android.widget.LinearLayout
+import androidx.annotation.RequiresApi
+import androidx.appcompat.app.AppCompatActivity
+import com.bitmovin.player.PlayerView
+import com.bitmovin.player.api.Player
+import com.bitmovin.player.api.ui.PictureInPictureHandler
+import com.facebook.react.bridge.LifecycleEventListener
+import com.facebook.react.bridge.ReactApplicationContext
+
+@SuppressLint("ViewConstructor")
+open class RNBasePlayerView(val context: ReactApplicationContext): LinearLayout(context),
+    LifecycleEventListener, PictureInPictureHandler, View.OnLayoutChangeListener {
```

Comment:
Actually, I created this `RNBasePlayerView` with the intention of removing this PiP handling logic from the `RNPlayerView`, which was getting really bloated. But I wasn't so sure how I should extract such behavior from it, so I just moved it to a parent class even though this is not the best solution 😅.

I'll try to extract the PiP handling into its own component (without the whole view logic) and let `RNPlayerView` use it directly. This way we can get rid of this base class and unnecessary inheritances too.

Reply:
Thanks for the explanation, now I understand why you decided for the given design.

That sounds good 👍 The bitmovin android player library also provides a `DefaultPictureInPictureHandler` (you probably have seen it), but it is just a minimalistic variant that might fit for simple cases without any other UI components i.e. it does a bit less than what you have implemented already.
But still I think you could extract everything into your own dedicated PiP handler and just instantiate it directly in the view.
It would be cool if we could inject it from outside, so we are more flexible, but I am not sure if that's possible/feasible in this RN wrapper.

Reply:
Yes, I checked the `DefaultPictureInPictureHandler` and even used it in my initial implementation, but I noticed it has a few glitches when animating PiP transitions. It basically lacks a few Android parameters described in [the docs](https://developer.android.com/develop/ui/views/picture-in-picture) to enable smoother transitions, especially for video content. So I decided to create a custom `PictureInPictureHandler` to include such parameters when available and provide the best possible PiP experience for the user's running version.

Also, I can try to make it injectable from the outside e.g. from other RN native modules, but I don't think we should focus on making it injectable from JavaScript code right now since it would require the lib to expose a lot of Android-specific APIs and considerably increase the complexity of this PR. If you're interested, we could schedule a kick-off meeting later just to discuss how this Android `PictureInPictureHandler` API could look like from JS and implement it in a future PR. What do you think?

Reply:
I have moved RN pip handler to its own component and made it injectable from other native modules. See the commits: 452c4c2da3665ac877b026b32424529e8af0ab61,  e9c8533bec1fb59a3fed0836014acb88fd35b81b.

Reply:
> I don't think we should focus on making it injectable from JavaScript code right now

Perfect! That was also not my intention, PiP feels quite OS specific, I was not sure if it is super nice to handle this from JS, but you kind of confirmed that it is not too nice anyway :) 

Regarding a sync: What do you think would be the benefit with a JS PiP handler. What use cases would we enabled with that?

Reply:
I think the only benefit of handling PiP from JS would be to minimize the amount of native code the users have to write and maintain themselves. It’s usually better to keep as much app logic in JS as possible, at least from the user perspective, since it helps with testing and portability between RN versions. Other than that, I can’t see many benefits.